src/api/question/question.service.ts

```diff
@@ -0,0 +1,176 @@
+import { Injectable } from '@nestjs/common'
+import { CreateQuestionDto } from './dto/create-question.dto'
+import { UpdateQuestionDto } from './dto/update-question.dto'
+import { QuestionEntity } from '@api/question/entities/question.entity'
+import { Repository } from 'typeorm'
+import { InjectRepository } from '@nestjs/typeorm'
+import { ValidationException } from '@exceptions/validation.exception'
+import { ErrorCode } from '@constants/error-code.constant'
+import { LessonEntity } from '@api/lesson/entities/lesson.entity'
+import { AnswerEntity } from '@api/answer/entities/answer.entity'
+import { plainToInstance } from 'class-transformer'
+import { QuestionResponseDto, CreateQuestionResponseDto } from './dto/question-response.dto'
+import { QuestionType } from '@common/enums/question.enum'
+import { paginate } from '@utils/offset-pagination'
+import { ListQuestionReqDto } from '@api/question/dto/list-question.req.dto'
+
+@Injectable()
+export class QuestionService {
+  constructor(
+    @InjectRepository(QuestionEntity)
+    private readonly questionRepository: Repository<QuestionEntity>,
+
+    @InjectRepository(LessonEntity)
+    private readonly lessonRepository: Repository<LessonEntity>,
+
+    @InjectRepository(AnswerEntity)
+    private readonly answerRepository: Repository<AnswerEntity>
+  ) {}
+  async create(createQuestionDto: CreateQuestionDto): Promise<CreateQuestionResponseDto> {
+    try {
+      // 1. Check if the lesson exists
+      const lesson = await this.lessonRepository.findOne({ where: { lessonId: createQuestionDto.lesson_id } })
+      if (!lesson) {
+        throw new ValidationException(ErrorCode.L001, 'Lesson not found')
+      }
+
+      const newQuestion = this.questionRepository.create({
+        lessonId: createQuestionDto.lesson_id,
+        content: createQuestionDto.content,
+        type: createQuestionDto.type,
+        difficulty: createQuestionDto.difficulty,
+        additionalImage: createQuestionDto.additional_image
+      })
+
+      const savedQuestion = await this.questionRepository.save(newQuestion)
+
+      // 2. Create answers for this question
+
+      // 2.1 Check if the question type is Multiple
+      if (createQuestionDto.type === QuestionType.MULTI_CHOICE) {
+        if (createQuestionDto.answers.length !== 4) {
```

Comment:
I think we should not use the current validation as there are questions may have more options than 4
How about
```suggestion
        if (createQuestionDto.answers.length <= 2) {
```